ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/electra/block/BlockProcessorElectra.java

```diff
@@ -335,6 +344,105 @@ public void processDepositReceipts(
+          electraState.getPendingConsolidations().size()
+              < specConfigElectra.getPendingConsolidationsLimit(),
+          "Consolidation queue is full");
+
+      // If there is too little available consolidation churn limit, no consolidations are allowed
+      // in the block
+      assertCondition(
+          BeaconStateAccessorsElectra.required(beaconStateAccessors)
+              .getConsolidationChurnLimit(electraState)
+              .isGreaterThan(specConfigElectra.getMinActivationBalance()),
+          "Not enough consolidation churn limit available");
+
+      final Consolidation consolidation = signedConsolidation.getMessage();
+
+      // Verify that source != target, so a consolidation cannot be used as an exit.
+      assertCondition(
+          consolidation.getSourceIndex() != consolidation.getTargetIndex(),
+          "Consolidation source_index and target_index must be different");
+
+      final Validator sourceValidator = state.getValidators().get(consolidation.getSourceIndex());
+      final Validator targetValidator = state.getValidators().get(consolidation.getTargetIndex());
+      final UInt64 currentEpoch = beaconStateAccessors.getCurrentEpoch(electraState);
+
+      assertCondition(
+          predicatesElectra.isActiveValidator(sourceValidator, currentEpoch),
+          "Source validator in consolidation is inactive");
+      assertCondition(
+          predicatesElectra.isActiveValidator(targetValidator, currentEpoch),
+          "Target validator in consolidation is inactive");
+
+      // Verify exits for source and target have not been initiated
+      assertCondition(
+          sourceValidator.getExitEpoch().equals(FAR_FUTURE_EPOCH),
+          "Source validator in consolidation is exiting");
+      assertCondition(
+          targetValidator.getExitEpoch().equals(FAR_FUTURE_EPOCH),
+          "Target validator in consolidation is exiting");
+
+      // Consolidations must specify an epoch when they become valid; they are not valid before then
+      assertCondition(
+          currentEpoch.isGreaterThanOrEqualTo(consolidation.getEpoch()),
+          "Consolidation must have activation epoch in the future");
+
+      // Verify the source and the target have Execution layer withdrawal credentials
+      assertCondition(
+          predicatesElectra.hasExecutionWithdrawalCredential(sourceValidator),
+          "Source validator in consolidation must have execution withdrawal credentials");
+      assertCondition(
+          predicatesElectra.hasExecutionWithdrawalCredential(targetValidator),
+          "Target validator in consolidation must have execution withdrawal credentials");
+      // Verify the same withdrawal address
+      assertCondition(
+          sourceValidator
+              .getWithdrawalCredentials()
+              .equals(targetValidator.getWithdrawalCredentials()),
+          "Source and Target validators in consolidation must have the same withdrawal credentials");
```

Comment:
this is more complicated, comparing the eth address portion of it only - it may be worth having a helper that just extracts that (we might have one...)
the reason is target may be already 0x02 so in that case this check would fail